hstream/src/HStream/Client/Types.hs

```diff
@@ -194,23 +197,25 @@ data ShardOffset = EARLIEST
                  | Timestamp Int64
   deriving (Show)
 
-instance Read ShardOffset where
-  readPrec = do
-    l <- Read.lexP
-    case l of
-      Read.Ident "earliest" -> return EARLIEST
-      Read.Ident "latest" -> return LATEST
-      Read.Ident "timestamp" -> do Read.Symbol ":" <- Read.lexP
-                                   t :: Int64 <- Read.readPrec
-                                   return $ Timestamp t
-      Read.Ident "recordId" -> do Read.Symbol ":" <- Read.lexP
-                                  sId :: Word64 <- Read.readPrec
-                                  Read.Symbol "-" <- Read.lexP
-                                  bId :: Word64 <- Read.readPrec
-                                  Read.Symbol "-" <- Read.lexP
-                                  idx :: Word32 <- Read.readPrec
-                                  return $ RecordId sId bId idx
-      x -> errorWithoutStackTrace $ "cannot parse StatsCategory: " <> show x
+offsetParser :: ReadM ShardOffset
+offsetParser = eitherReader (parseShardOffset . T.pack)
```

Comment:
This is not actually a parser. Using `offsetReader` is better than `offsetParser`

```suggestion
offsetReader :: ReadM ShardOffset
offsetReader = eitherReader (parseShardOffset . T.pack)
```